presto-spi/src/main/java/com/facebook/presto/spi/plan/SequenceNode.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.plan;
+
+import com.facebook.presto.spi.SourceLocation;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class SequenceNode
+        extends PlanNode
+{
+    private final List<PlanNode> leftSources;
+    private final PlanNode right;
+
+    @JsonCreator
+    public SequenceNode(Optional<SourceLocation> sourceLocation,
+            @JsonProperty("id") PlanNodeId planNodeId,
+            @JsonProperty("leftPlanList") List<PlanNode> left,
+            @JsonProperty("right") PlanNode right)
+    {
+        this(sourceLocation, planNodeId, Optional.empty(), left, right);
+    }
+
+    public SequenceNode(Optional<SourceLocation> sourceLocation,
+            PlanNodeId planNodeId,
+            Optional<PlanNode> statsEquivalentPlanNode,
+            List<PlanNode> leftList,
+            PlanNode right)
+    {
+        super(sourceLocation, planNodeId, statsEquivalentPlanNode);
+        this.leftSources = leftList;
+        this.right = right;
+    }
+
+    @JsonProperty
+    public List<PlanNode> getLeftSources()
+    {
+        return this.leftSources;
+    }
+
+    @JsonProperty
+    public PlanNode getRight()
```

Comment:
I think the names leftSource and right are a bit confusing.
Open to suggestion on how to name these better, but perhaps something like this:
- leftSources -> cteDefinitions or cteProducers
- right -> ???

Reply:
Agreed

Reply:
> right -> source
This might be confusing since getSources needs to return leftsources + right. Maybe primarySource?

Reply:
primarySource sounds good to me